src/main/java/com/commercetools/project/sync/CliRunner.java

```diff
@@ -201,9 +216,40 @@ private static Options buildCliOptions() {
     final boolean isFullSync = commandLine.hasOption(FULL_SYNC_OPTION_SHORT);
     final boolean isSyncProjectSyncCustomObjects =
         commandLine.hasOption(SYNC_PROJECT_SYNC_CUSTOM_OBJECTS_OPTION_LONG);
+    final boolean isProductSyncConfigOptionPresent =
+        commandLine.hasOption(PRODUCT_SYNC_CONFIG_OPTION);
 
+    final ProductSyncCustomRequest productSyncCustomRequest;
+    try {
+      productSyncCustomRequest =
+          isProductSyncConfigOptionPresent
+              ? parseProductSyncConfigOption(commandLine.getOptionValue(PRODUCT_SYNC_CONFIG_OPTION))
+              : null;
+
+    } catch (CliException e) {
+      return exceptionallyCompletedFuture(e);
+    }
     return syncerFactory.sync(
-        syncOptionValues, runnerNameValue, isFullSync, isSyncProjectSyncCustomObjects);
+        syncOptionValues,
+        runnerNameValue,
+        isFullSync,
+        isSyncProjectSyncCustomObjects,
+        productSyncCustomRequest);
+  }
+
+  public static ProductSyncCustomRequest parseProductSyncConfigOption(String customRequest) {
```

Comment:
Imo this parse function would be a part of the object, maybe take the passed string as a constructor and only return getters from ProductSyncCustomRequest? Wdyt?

Reply:
https://github.com/commercetools/commercetools-project-sync/pull/243/commits/bcd7bb9b21c50bea0dc0b0d76ba8000e12be3361.  I moved this static method to ProductSyncCustomRequest class. I think this should be good. Wdyt?

Reply:
I would rather use the type and constructor and pass ProductSyncCustomRequest to the ProductSyncer not as a string but as an object.

Reply:
https://github.com/commercetools/commercetools-project-sync/pull/243/commits/3f76c4af0233b0d433c9dafaa99048daf1ae01c7